controllers/storagecluster/cephfilesystem.go

```diff
@@ -108,6 +135,40 @@ func (r *StorageClusterReconciler) newCephFilesystemInstances(initStorageCluster
 	return []*cephv1.CephFilesystem{ret}, nil
 }
 
+// getSubVolumeGroupsForFilesystem returns the subVolumeGroups for the given filesystem
+func getSubVolumeGroupsForFilesystem(filesystem *cephv1.CephFilesystem) []cephv1.CephFilesystemSubVolumeGroup {
+	// Default value of "distributed" option for pinning in the CephFilesystemSubVolumeGroup CR
+	var defaultPinningValue = 1
+	subVolumeGroupSpec := cephv1.CephFilesystemSubVolumeGroupSpec{
+		FilesystemName: filesystem.Name,
+		Pinning: cephv1.CephFilesystemSubVolumeGroupSpecPinning{
+			Distributed: &defaultPinningValue,
+		},
+	}
+	defaultSubVolumeGroup := cephv1.CephFilesystemSubVolumeGroup{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      defaultSubvolumeGroupName,
+			Namespace: filesystem.Namespace,
+		},
+		Spec: subVolumeGroupSpec,
+	}
+	subVolumeGroups := []cephv1.CephFilesystemSubVolumeGroup{defaultSubVolumeGroup}
+	for i := range filesystem.Spec.DataPools {
+		if i == 0 {
+			continue // Skip the first data pool as for it the default svg is already added
+		}
+		svg := cephv1.CephFilesystemSubVolumeGroup{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      generateNameForAdditionalCephFilesystemStorageClassSubvolumeGroup(filesystem.Name, i),
+				Namespace: filesystem.Namespace,
+			},
+			Spec: subVolumeGroupSpec,
+		}
+		subVolumeGroups = append(subVolumeGroups, svg)
+	}
+	return subVolumeGroups
```

Comment:
Why do we create a subvolume group per data pool by default? A sub-volume group requirement is orthogonal to the number of data pools and it is used to segregate access not to designate storage location.